File: examples/dotnet/HelloPlugin/Content/ThunderWand.cs
using Falcon;

namespace Hello.Content
{
    public static class ThunderWand
    {
        public const string ItemIdentifier = "hello:thunder_wand";

        public static void RegisterTo(Plugin plugin)
        {
            CustomItem wand = new CustomItem
            {
                Identifier = ItemIdentifier,
                DisplayName = "Thunder Wand",
                Icon = "thunder_wand",
                CreativeCategory = "equipment",
                MaxStackSize = 1,
                HandEquipped = true,
                OnUseOnBlock = StrikeLightning
            };

            if (!plugin.Content.AddItem(wand))
            {
                plugin.Logger.Warning("Could not register the thunder wand");
            }
        }

        private static bool StrikeLightning(Player player, Item item, BlockPos position, uint face)
        {
            player.Entity.Level.StrikeLightning(position.Center);
            return true;
        }
    }
}

File: examples/java/HelloPlugin/src/main/java/falcon/examples/hello/content/ThunderWand.java
package falcon.examples.hello.content;

import falcon.api.Plugin;
import falcon.api.content.CustomItem;

public final class ThunderWand {
    public static final String IDENTIFIER = "hello:thunder_wand";

    private ThunderWand() {
    }

    public static void registerTo(Plugin plugin) {
        CustomItem wand = new CustomItem(IDENTIFIER)
                .displayName("Thunder Wand")
                .icon("thunder_wand")
                .creativeCategory("equipment")
                .maxStackSize(1)
                .handEquipped(true)
                .onUseOnBlock((player, item, position, face) -> {
                    player.entity().level().strikeLightning(position.center());
                    return true;
                });

        if (!plugin.content().item(wand)) {
            plugin.logger().warning("Could not register the thunder wand");
        }
    }
}
